src/Hosting/Hosting/src/Internal/HostingTelemetryHelpers.cs

```diff
@@ -0,0 +1,132 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Frozen;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Hosting;
+
+internal static class HostingTelemetryHelpers
+{
+    // Semantic Conventions for HTTP.
+    // Note: Not all telemetry code is using these const attribute names yet.
+    public const string AttributeHttpRequestMethod = "http.request.method";
+    public const string AttributeHttpRequestMethodOriginal = "http.request.method_original";
+    public const string AttributeUrlScheme = "url.scheme";
+    public const string AttributeUrlPath = "url.path";
+    public const string AttributeServerAddress = "server.address";
+    public const string AttributeServerPort = "server.port";
+    public const string AttributeUserAgentOriginal = "user_agent.original";
+
+    // The value "_OTHER" is used for non-standard HTTP methods.
+    // https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md#common-attributes
+    private const string OtherHttpMethod = "_OTHER";
+
+    private static readonly object[] BoxedStatusCodes = new object[512];
+
+    private static readonly FrozenDictionary<string, string> KnownHttpMethods = FrozenDictionary.ToFrozenDictionary([
+        KeyValuePair.Create(HttpMethods.Connect, HttpMethods.Connect),
+        KeyValuePair.Create(HttpMethods.Delete, HttpMethods.Delete),
+        KeyValuePair.Create(HttpMethods.Get, HttpMethods.Get),
+        KeyValuePair.Create(HttpMethods.Head, HttpMethods.Head),
+        KeyValuePair.Create(HttpMethods.Options, HttpMethods.Options),
+        KeyValuePair.Create(HttpMethods.Patch, HttpMethods.Patch),
+        KeyValuePair.Create(HttpMethods.Post, HttpMethods.Post),
+        KeyValuePair.Create(HttpMethods.Put, HttpMethods.Put),
+        KeyValuePair.Create(HttpMethods.Trace, HttpMethods.Trace)
+    ], StringComparer.OrdinalIgnoreCase);
```

Comment:
Add `QUERY` given #63260 went in yesterday?

Reply:
This list is based on the known values list at https://opentelemetry.io/docs/specs/semconv/registry/attributes/http

<img width="1317" height="849" alt="image" src="https://github.com/user-attachments/assets/d1ce3ec3-fa56-41a5-bb59-21ba96381708" />

You're welcome to get OTEL folks to add QUERY and then we'll add it here.